src/main/java/org/alliancegenome/curation_api/dao/ConstructDAO.java

```diff
@@ -1,15 +1,42 @@
 package org.alliancegenome.curation_api.dao;
 
+import jakarta.persistence.Query;
 import org.alliancegenome.curation_api.dao.base.BaseSQLDAO;
 import org.alliancegenome.curation_api.model.entities.Construct;
 
 import jakarta.enterprise.context.ApplicationScoped;
 
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 @ApplicationScoped
 public class ConstructDAO extends BaseSQLDAO<Construct> {
 	
 	protected ConstructDAO() {
 		super(Construct.class);
 	}
 
+	public Map<String, Long> getConstructIdMap() {
+		if (constructIdMap.size() > 0) {
+			return constructIdMap;
+		}
+		Query q = entityManager.createNativeQuery("SELECT a.id, a.modEntityId, a.modInternalId FROM Reagent as a where exists (select * from construct as g where g.id = a.id)");
+		List<Object[]> ids = q.getResultList();
+		ids.forEach(record -> {
+			if (record[1] != null) {
+				constructIdMap.put((String) record[1], (long)record[0]);
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'typecast' is not followed by whitespace.
<!-- __reviewdog__:ChA5YzFjYzljNjYyMTdiMDQ0EgpjaGVja3N0eWxl -->